src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServices
+{
+    internal abstract class AbstractSyntaxFactsService
+    {
+        private readonly static ObjectPool<List<Dictionary<string, string>>> s_aliasMapListPool =
+            new ObjectPool<List<Dictionary<string, string>>>(() => new List<Dictionary<string, string>>());
+        private readonly static ObjectPool<Dictionary<string, string>> s_aliasMapPool =
+            new ObjectPool<Dictionary<string, string>>(() => new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase));
```

Comment:
We store all this stuff case insensitively.  It means we may get *more* hits (though that will be extremely rare in practice).  All that means is we'll end up resolving more stuff in teh DependentTypeFinder.  That's fine though.